javascript/testing_javascript/project_battleship.md

```diff
@@ -9,27 +9,31 @@ We have not yet discussed testing the appearance of a webpage, which requires a
-   1. We'll leave the HTML implementation up to you for now, but you should display both the player's boards and render them using information from the Gameboard class/factory.
-      1. You need methods to render the gameboards and to take user input for attacking.  For attacks, let the user click on a coordinate in the enemy Gameboard.
-   1. The game loop should step through the game turn by turn using only methods from other objects.  If at any point you are tempted to write a new function inside the game loop, step back and figure out which class or module that function should belong to.
-   1. Create conditions so that the game ends once one player's ships have all been sunk.  This function is appropriate for the Game module.
+1. Create `Player`
+    1. There will be two types of players in the game, 'real' players and 'computer' players.
+    1. Each player object should contain it's own gameboard.
+1. Import your classes/factories into another file, and drive the game using event listeners to interact with your objects. Create a module that helps you manage actions that should happen in the DOM
+    1. At this point it is appropriate to begin crafting your User Interface.
+    1. Set up a new game by creating Players. For now just populate each player's Gameboard with predetermined coordinates. You are going to implement a system for allowing players to place their ships later.
+    1. We’ll leave the HTML implementation up to you for now, but you should display both the player’s boards and render them using information from the Gameboard class/factory.
+        1. You'll need methods to render each player's Gameboard, so put them in an appropriate module.
+    1. Your event listeners should step through the game turn by turn using only methods from other objects. If at any point you are tempted to write a new function, step back and figure out which class or module that function should belong to.
+    1. For attacks, let the user click on a coordinate in the enemy Gameboard. Send the user input to methods on your objects, and re-render the boards to display the new information.
+        1. Players should take turns playing the game by attacking the enemy Gameboard. If you feel the need to keep track of the current player's turn, it's appropriate to manage that in this module, instead of another mentioned object.
+        1. The game is played against the computer, so make the ‘computer’ players capable of making random plays. The AI does not have to be smart, but it should know whether or not a given move is legal (i.e. it shouldn’t shoot the same coordinate twice).
+    1. Create conditions so that the game ends once one player’s ships have all been sunk. This function is also appropriate for this module.
 1. Finish it up
    1. Implement the system that allows players to place their ships. You can let them type coordinates for each ship or have a button to cycle through random placements.
```

Comment:
```suggestion
1. Finish it up by implementing a system that allows players to place their ships. For example, you can let them type coordinates for each ship or have a button to cycle through random placements.
```
While we're at it, I think this would be better as just a single top-level item?

Reply:
Sounds good to me